src/fetchPost.js

```diff
@@ -0,0 +1,365 @@
+            !hasImage ? "hidden" : ""
+          }">
+            ${images
+              .map(
+                (image) => `
+            <section class="single-post__thumbnail square-image-container">
+            <div class="square-image-container__wrapper">
+              <img
+                class="square-image-container__image"
+                src="${image.image.url}"
+              />
+            </div>
+            </section>`
+              )
+              .join(" ")}
+          </section>
+          <section class="single-post__note-container">
+            <textarea id="note" class="single-post__note" type="textbox" placeholder="یادداشت">${
+              notes ? (notes[postID] ? notes[postID] : "") : ""
+            }</textarea>
+            <section class="single-post__note-warning">یادداشت تنها برای شما قابل دیدن است و پس از حذف آگهی، پاک خواهد شد.
+            </section>
+          </section>
+          <a class="single-post__safe-buy iconed-link" href="https://support.divar.ir/b/support-users/fa/kb/articles/article-46">
+                <img class="iconed-link__icon" src="${
+                  require("./icons/svgs/check-circle-o.svg").default
+                }" />
+                <span class="iconed-link__text">راهنمای خرید امن</span>
+          </a>
+          <hr class="single-post__row-separator"/>
+          <a class="single-post__report iconed-link" href="https://support.divar.ir/b/support-users/fa/kb/articles/article-46">
+                <img class="iconed-link__icon" src="${
+                  require("./icons/svgs/info-circle-o.svg").default
+                }" />
+                <span class="iconed-link__text">ثبت تخلف و مشکل آگهی</span>
+          </a>
+          <hr class="single-post__row-separator"/>
+        </section>
+      </article>
+      </main>`.concat(container.innerHTML);
+
+      //carousol handler
+      let res = document.querySelectorAll(".single-post__thumbnail");
+      res.forEach((element) => {
+        element.onclick = function (event) {
+          const image = document.querySelector(".single-post__image");
+          const newImage = event.target.src;
+          image.src = newImage;
+        };
+      });
+
+      //bookmark handler
+      document.querySelector(".single-post__action-bookmark").onclick =
+        function () {
+          let bookmarks = JSON.parse(window.localStorage.getItem("bookmarks"));
+          let icon = document.querySelector(".single-post__bookmark-icon");
+          let bookmarked = false;
+
+          if (bookmarks) {
+            for (let i = 0; i < bookmarks.length; i++) {
+              if (bookmarks[i].url == url) {
+                bookmarked = true;
+              }
+            }
+          }
+
+          if (!icon.classList.contains("bookmarked")) {
+            icon.src = require("./icons/svgs/bookmarked.svg").default;
+            icon.classList.toggle("bookmarked");
+            if (!bookmarked) {
```

Comment:
You have two sources of truth here, one is the existence of the URL in local storage, and the other one is existence of bookmarked classname on the element. You should only rely on one of them and sync the other one with it.  
My approach would've been to rely on the bookmarked from localStorage, and then do all of this logic based on that. Then I could've simply used `icon.classList.remove` and `icon.classList.add` instead of toggle.